text/0002-sarif-result-schema.md

```diff
@@ -0,0 +1,1939 @@
+          "Total eslint-disable usages: 32"
+        ],
+        "input": 32
+      },
+      {
+        "name": "reduce-outdated-major-dependencies",
+        "summary": "Update outdated major versions",
+        "details": "28 major versions outdated",
+        "defaultThreshold": 0.05,
+        "items": [],
+        "input": 0.24347826086956523
+      },
+      {
+        "name": "reduce-outdated-minor-dependencies",
+        "summary": "Update outdated minor versions",
+        "details": "9 minor versions outdated",
+        "defaultThreshold": 0.05,
+        "items": [],
+        "input": 0.0782608695652174
+      },
+      {
+        "name": "reduce-outdated-dependencies",
+        "summary": "Update outdated versions",
+        "details": "32% of versions outdated",
+        "defaultThreshold": 0.2,
+        "items": [],
+        "input": 0.3217391304347826
+      }
+    ]
+  },
+  "runs": [
+    {
+      "results": [
+        {
+          "message": {
+            "text": "components"
+          },
+          "locations": [],
+          "occurrenceCount": 219,
+          "properties": {
+            "taskName": "ember-types",
+            "taskDisplayName": "Ember Types",
+            "category": "metrics",
+            "group": "ember"
+          }
+        },
+        {
+          "message": {
+            "text": "controllers"
+          },
+          "locations": [],
+          "occurrenceCount": 41,
+          "properties": {
+            "taskName": "ember-types",
+            "taskDisplayName": "Ember Types",
+            "category": "metrics",
+            "group": "ember"
+          }
+        },
+        {
+          "message": {
+            "text": "helpers"
+          },
+          "locations": [],
+          "occurrenceCount": 35,
+          "properties": {
+            "taskName": "ember-types",
+            "taskDisplayName": "Ember Types",
+            "category": "metrics",
+            "group": "ember"
+          }
+        },
+        {
+          "message": {
+            "text": "initializers"
+          },
+          "locations": [
+            {
+              "physicalLocation": {
+                "artifactLocation": {
+                  "uri": "/app/initializers/app.js"
+                }
+              }
+            },
+            {
+              "physicalLocation": {
+                "artifactLocation": {
+                  "uri": "/app/initializers/configure-inflector.js"
+                }
+              }
+            },
+            {
+              "physicalLocation": {
+                "artifactLocation": {
+                  "uri": "/app/initializers/services.js"
+                }
+              }
+            },
+            {
+              "physicalLocation": {
+                "artifactLocation": {
+                  "uri": "/app/initializers/tracer.js"
+                }
+              }
+            }
+          ],
+          "occurrenceCount": 4,
+          "properties": {
```

Comment:
The challenge with stuffing things into a property bag like this is you need intimate knowledge of the contents of the property bag in order to access it. It's not easily iterable, and it's hard to write dynamic loops that try to generically access this data. WDYT?

Reply:
We discussed this a bit offline; I agree with the risks/problems with this, but I think the pros of using this schema as a whole outweigh the cons, and I don't see another option for this type of  data with SARIF. WDYT?

Reply:
I'd tend to agree. That's the price we're paying with using a standardized format - we have to accommodate the fact that it's general purpose.